judgels-frontends/raphael/src/components/ProblemWorksheetCard/Bundle/ProblemStatementCard/ItemEssayCard/ItemEssayForm/ItemEssayForm.tsx

```diff
@@ -88,11 +94,74 @@ export default class ItemEssayForm extends React.PureComponent<ItemEssayFormProp
             Answered.
           </Callout>
         );
+      case AnswerState.ClearingAnswer:
+        return (
+          <Callout intent={Intent.NONE} icon="ban-circle" className="essay-callout">
+            Clearing answer...
+          </Callout>
+        );
       default:
         return <div className="bp3-callout bp3-callout-icon essay-callout-edit">&nbsp;</div>;
     }
   }
 
+  renderClearAnswerButton() {
+    return (
+      (this.state.answerState === AnswerState.AnswerSaved || this.state.answerState === AnswerState.ClearingAnswer) && (
+        <Button
+          type="button"
+          text={StatementButtonText.ClearAnswer}
+          intent={Intent.DANGER}
+          onClick={this.onClearAnswerButtonClick}
+          className="essay-button"
+        />
+      )
+    );
+  }
+
+  renderClearAnswerDialog() {
```

Comment:
This is not necessary. We can just simply use native https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm. Will greatly reduce code.